src/org/parosproxy/paros/extension/state/ExtensionState.java

```diff
@@ -67,11 +75,46 @@ public void hook(ExtensionHook extensionHook) {
 	    // ZAP: Changed to check if there is a view.
 	    if (getView() != null) {
 		    extensionHook.getHookMenu().addEditMenuItem(extensionHook.getHookMenu().getMenuSeparator());
+		    extensionHook.getHookMenu().addEditMenuItem(getMenuZAPMode());
 		    extensionHook.getHookMenu().addEditMenuItem(getMenuSessionTrackingEnable());
 		    extensionHook.getHookMenu().addEditMenuItem(getMenuResetSessionState());
 	    }
 	}
 
+	private JMenuItem getMenuZAPMode(){
+		if (menuZAPMode == null) {
+			menuZAPMode = new JMenu(Constant.messages.getString("menu.edit.zapmode"));
+			int modeCount = 0;
+			for (Mode modeType : Mode.values()) {
+			menuItemsMode[modeCount] = addZAPModeMenuItem(modeType);
+  		menuZAPMode.add(menuItemsMode[modeCount]);
+			modeCount++;
+			}
+			modeCount = 0;
+			for(Mode modeType : Mode.values()){
+			addZAPModeEventListeners(modeType,menuItemsMode[modeCount],menuItemsMode);

```

Comment:
It could be used a `ButtonGroup` instead, which already takes care to keep just one of the buttons selected.